js/src/common/components/LoadingIndicator.tsx

```diff
@@ -0,0 +1,56 @@
+import Component, { ComponentAttrs } from '../Component';
+import classList from '../utils/classList';
+
+interface Attrs {
```

Comment:
Shouldn't this extend ComponentAttrs?

Reply:
No, the declaration of Component accepts a list of Attrs, which then extends upon ComponentAttrs, so it handles all that for us.

Reply:
`export default abstract class Component<T extends ComponentAttrs = ComponentAttrs> implements Mithril.ClassComponent<T>`

But `T` here needs to extend `ComponentAttrs`

Reply:
And we'd want to export the interface of attributes as well.

Reply:
@askvortsov1 My bad -- I misunderstood the way TS applies types like this! Thanks for teaching me :P